securedrop_client/gui/widgets.py

```diff
@@ -1132,27 +1414,24 @@ def set_snippet(self, source_uuid: str, collection_uuid: str = None, content: st
         if source_uuid != self.source_uuid:
             return
 
-        if not self.source.server_collection:
-            return
-
-        last_activity = self.source.server_collection[-1]
-        if collection_uuid and collection_uuid != last_activity.uuid:
-            return
+        if self.deleting:
+            content = ""
+        elif not self.source.server_collection:
+            if self.source.interaction_count > 0:
+                content = _("\u2014 All files and messages deleted for this source \u2014")
```

Comment:
A code comment should be added to explain that the `interaction_count` is never reset or decremented when source conversation items are deleted which is why we're able to use it to determine whether or not a source with zero conversation items has ever had a conversation with a journalist before.